celery/concurrency/asynpool.py

```diff
@@ -1085,6 +1086,11 @@ def on_job_process_lost(self, job, pid, exitcode):
         was assigned to exited by mysterious means (error exitcodes and
         signals).
         """
+        # if exit with EX_OK, won't mark it as worker lost,
+        # it's just normal (warm) shutdown
+        if exitcode == EX_OK and job.ready():
```

Comment:
Instead of the comment, maybe we should use a helper method?
It would be clearer.